packages/components/src/DateTimePickers/pickers/DatePicker/DatePicker.component.js

```diff
@@ -92,63 +105,61 @@ class DatePicker extends React.PureComponent {
-									const selected = this.isSelectedDate(date);
-									const today = isToday(date);
-									const shouldBeFocussable =
-										(selectedInCurrentCalendar && selected) ||
-										(!selectedInCurrentCalendar && day === 1);
-
-									const className = classNames(
-										theme['calendar-day'],
-										{
-											[theme.selected]: selected,
-											[theme.today]: today,
-										},
-										'tc-date-picker-day',
-										'btn-tertiary',
-										'btn-default',
-									);
-
-									let ariaLabel = format(date, 'dddd DD MMMM YYYY', pickerLocale);
-									const tdProps = {
-										key: j,
-										className: theme['calendar-col'],
-									};
-									if (selected) {
-										tdProps['aria-current'] = 'date';
-										ariaLabel = t('DATEPICKER_DAY_SELECTED', {
-											defaultValue: '{{date}}, selected',
-											date: ariaLabel,
-										});
-									}
-									if (today) {
-										ariaLabel = t('DATEPICKER_DAY_TODAY', {
-											defaultValue: 'Today, {{date}}',
-											date: ariaLabel,
-										});
-									}
-									return (
-										<td {...tdProps}>
-											<button
-												type="button"
-												className={className}
-												onClick={event => {
-													this.props.onSelect(event, date);
-												}}
-												disabled={disabled}
-												tabIndex={this.props.allowFocus && shouldBeFocussable ? 0 : -1}
-												onKeyDown={event => this.props.onKeyDown(event, this.calendarRef, day - 1)}
-												aria-label={ariaLabel}
-											>
-												{day}
-											</button>
-										</td>
-									);
+								const day = getDate(date);
+								const disabled = this.isDisabledDate(date);
+								const selected = this.isSelectedDate(date);
+								const today = isToday(date);
+								const shouldBeFocussable =
+									(selectedInCurrentCalendar && selected) ||
+									(!selectedInCurrentCalendar && day === 1);
+
+								const className = classNames(
+									theme['calendar-day'],
+									{
+										[theme.selected]: selected,
+										[theme.today]: today,
+										[theme['not-current-month']]: !this.isCurrentMonth(date),
+									},
+									'tc-date-picker-day',
+									'btn-tertiary',
+									'btn-default',
+								);
+
+								let ariaLabel = format(date, 'dddd DD MMMM YYYY', pickerLocale);
+								const tdProps = {
+									key: j,
+									className: theme['calendar-col'],
+								};
+								if (selected) {
+									tdProps['aria-current'] = 'date';
+									ariaLabel = t('DATEPICKER_DAY_SELECTED', {
+										defaultValue: '{{date}}, selected',
+										date: ariaLabel,
+									});
+								}
+								if (today) {
+									ariaLabel = t('DATEPICKER_DAY_TODAY', {
+										defaultValue: 'Today, {{date}}',
+										date: ariaLabel,
+									});
 								}
-								return <td key={j} />;
+								const buttonProps = this.isCurrentMonth(date) ? { name: day } : undefined;
```

Comment:
Can you change the `name attribute` to a data-attribute which is more standard to add custom info.
For example `data-day`

Reply:
name is used for the key action, in code 
withCalendarGesture.js
function getAllItems(calendarRef) {
	return calendarRef.querySelectorAll('td > button[name]');
}
and this is used for date picker and month picker?
How about 'data-valid'?

Reply:
changed to be 'data-value'